src/libbson/src/bson/bson-atomic.h

```diff
@@ -47,42 +47,68 @@ enum bson_memory_order {
 #define MSVC_MEMORDER_SUFFIX(X)
 #endif
 
+#if defined(USE_LEGACY_GCC_ATOMICS) || (!defined(__clang__) && \
+   __GNUC__ == 4 && __GNUC_MINOR__ >= 1 && __GNUC_MINOR__ < 9)
+#define BSON_USE_LEGACY_GCC_ATOMICS
+#else
+#undef BSON_USE_LEGACY_GCC_ATOMICS
+#endif
+
+#ifdef BSON_USE_LEGACY_GCC_ATOMICS
+#undef BSON_IF_GNU_LIKE
+#define BSON_IF_GNU_LIKE(...)
+#define BSON_IF_GNU_LEGACY_ATOMICS(...) __VA_ARGS__
+#else
+#define BSON_IF_GNU_LEGACY_ATOMICS(...)
+#endif
```

Comment:
Could you add a comment on why `BSON_IF_GNU_LIKE` is redefined temporarily in this file, or change it (within this file) to `BSON_IF_GNU_NEW_ATOMICS` (or some other name). It is clear for now, but in the future I'm not sure it will be easy to understand.